artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java

```diff
@@ -309,16 +309,17 @@ public void run() {
     */
    @Override
    protected void performCachedLargeMessageDeletes() {
-      for (Long largeMsgId : largeMessagesToDelete) {
-         SequentialFile msg = createFileForLargeMessage(largeMsgId, LargeMessageExtension.DURABLE);
+      for (LargeServerMessage largeServerMessage : largeMessagesToDelete.values()) {
```

Comment:
Calling values actually creates a new List, if you're iterating the objects, simply call using forEach method on the collection.

Reply:
If you wish the collection itself to be iterable, then please add this functionality to LongConcurrentHashMap implementation, it shouldn;t be too hard, as already it has a forEach method

Reply:
@michaelandrepearce update, please review, Thanks.